milvus/Resource.ts

```diff
@@ -0,0 +1,227 @@
+   *
+   * #### Example
+   *
+   * ```
+   *  new milvusClient(MILUVS_ADDRESS).resourceManager.describeResrouceGroup({
+   *    resource_group: 'my-resource-group'
+   * });
+   * ```
+   */
+  async describeResourceGroup(
+    data: DesribeResourceGroupsReq
+  ): Promise<DescribeResourceGroupResponse> {
+    const promise = await promisify(this.client, 'DescribeResourceGroup', data);
+    return promise;
+  }
+
+  /**
+   * drop a resource group.
+   *
+   * @return
+   *  | Property | Type | Description |
+   *  | :-- | :-- | :-- |
+   *  | status.error_code | string | error code |
+   *  | status.reason | string | error reason |
+   *
+   * #### Example
+   *
+   * ```
+   *  new milvusClient(MILUVS_ADDRESS).resourceManager.dropResourceGroup({
+   *    resource_group: 'my-resource-group'
+   * });
+   * ```
+   */
+  async dropResourceGroup(data: DropResourceGroupsReq): Promise<ResStatus> {
+    const promise = await promisify(this.client, 'DropResourceGroup', data);
+    return promise;
+  }
+
+  /**
+   * transfer nodes from one resource group to another
+   *
+   * @return
+   *  | Property | Type | Description |
+   *  | :-- | :-- | :-- |
+   *  | status.error_code | string | error code |
+   *  | status.reason | string | error reason |
+   *
+   *
+   * #### Example
+   *
+   * ```
+   *  new milvusClient(MILUVS_ADDRESS).resourceManager.transferNode({
+   *    source_resource_group: 'source-resource-group',
+   *    target_resource_group: 'target-resource-group',
+   *    num_node: 2
+   * });
+   * ```
+   */
+  /* istanbul ignore next */
+  async transferReplica(data: TransferReplicaReq): Promise<ResStatus> {
+    const promise = await promisify(this.client, 'TransferReplica', data);
+    return promise;
+  }
+
+  /**
+   * transfer nodes from one resource group to another
+   *
+   * @return
+   *  | Property | Type | Description |
+   *  | :-- | :-- | :-- |
+   *  | status.error_code | string | error code |
+   *  | status.reason | string | error reason |
+   *
+   *
+   * #### Example
+   *
+   * ```
+   *  new milvusClient(MILUVS_ADDRESS).resourceManager.transferNode({
+   *    source_resource_group: 'source-resource-group',
+   *    target_resource_group: 'target-resource-group',
+   *    num_node: 4
+   * });
+   * ```
+   */
+  /* istanbul ignore next */
+  async transferNode(data: TransferNodeReq): Promise<ResStatus> {
+    const promise = await promisify(this.client, 'TransferNode', data);
+    return promise;
+  }
+
+  /**
+   * drop all resource groups, transfer all nodes to the default group
+   *
+   * @return
+   *  | Property | Type | Description |
+   *  | :-- | :-- | :-- |
+   *  | status.error_code | string | error code |
+   *  | status.reason | string[] | error reason |
+   *
+   *
+   * #### Example
+   *
+   * ```
+   *  new milvusClient(MILUVS_ADDRESS).resourceManager.dropResourceGroups();
+   * ```
+   */
+  async dropResourceGroups(): Promise<ResStatus[]> {
```

Comment:
Should we rename dropResourceGroups to dropAllResourceGroups ?
Or we can support users to specify which resources they want to drop.